blog/2022-04-25-jwt-vs-api-key-authentication/index.md

```diff
@@ -0,0 +1,109 @@
+
+When the client sends a request to the API it includes the JWT in the request's `Authorization` header. The API then validates the JWT to be authentic and can then use the information in the JWT to identify the client. Typically the JWT contains a `sub` parameter that identifies the client. The token also includes a `aud` parameter that specifies which API the token can call.
+
+JWT tokens can be issued with any length of expiration time, but it is typical for tokens to expire in a short period, such as one hour.
+
+JWT auth with OAuth uses the [Client Credentials](https://auth0.com/docs/get-started/authentication-and-authorization-flow/client-credentials-flow) flow on the identity server. Each client that will call the API is issued a `Client Id` and a `Client Secret` - think of these values like a username and password. The client uses these values to request an access token they use to call the API. In code, the client credentials flow looks like the following example.
+
+**Request**
+
+```bash
+curl --request POST \
+  --url 'https://YOUR_DOMAIN/oauth/token' \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --data grant_type=client_credentials \
+  --data client_id=YOUR_CLIENT_ID \
+  --data client_secret=YOUR_CLIENT_SECRET \
+  --data audience=YOUR_API_IDENTIFIER
+```
+
+**Response**
+
+```json
+{
+  "access_token": "eyJz93a...k4laUWw",
+  "token_type": "Bearer",
+  "expires_in": 86400
+}
+```
+
+### Considerations of Machine-to-Machine JWT Auth
+
+JWT-based API auth is a good choice for securing microservices, implementing services within an organization, or sharing APIs with certain types of external clients.
+
+- JWT tokens are typically not revokable. In order to prevent a JWT token from calling an API, you must roll the secrets of that client - this will disable ALL JWT tokens currently issued.
+- Permissions with JWT tokens are managed at the identity provider level, meaning that all tokens issued for the same client will have the same permissions.
+- JWT tokens are static; permissions, expiration time, or other properties cannot change once the token is issued.
+- JWT tokens should be kept relatively small in size (i.e., you don't want to put a list of all resources the client is authorized to access in the JTW token itself).
+- When JWT tokens expire, the consumer must request a new token using the Client ID and Secret value.
+- Identity Providers often charge based on the number of tokens issued.
+
+## API Key Authentication
+
+With API authentication, each client receives a unique secret key. Unlike JWT tokens, the key itself doesn't contain any actual data, it is simply a unique string associated with the client. Furthermore, there is no standard protocol for API Key authentication like OAuth, etc., so each implementation can differ.
+
+The simplest example of API token authentication is an environment variable that is checked against each API request.
```

Comment:
using token/key interchangeably, a bit confusing.

Reply:
do we need to show this example - I feel it trivializes API key auth a bit, we want it to sound scarier from the get-go